xarray/core/common.py

```diff
@@ -212,7 +212,7 @@ def __dir__(self):
         methods.
         """
         extra_attrs = [item for sublist in self._attr_sources
-                       for item in sublist]
+                       for item in sublist if isinstance(item, str)]
```

Comment:
basestring is a super class of both bytes and str on Python 2. On Python 3, it's only str.